pkgs/tools/security/ghidra/build.nix

```
@@ -116,13 +99,19 @@ in stdenv.mkDerivation {
   inherit pname version src;
 
   nativeBuildInputs = [
-    gradle unzip makeWrapper icoutils
+    gradle unzip makeWrapper icoutils protobuf
   ] ++ lib.optional stdenv.isDarwin xcbuild;
 
   dontStrip = true;
 
-  patches = [ ./0001-Use-protobuf-gradle-plugin.patch ];
-  postPatch = fixProtoc;
+  patches = [
+    ./0001-Use-protobuf-gradle-plugin.patch
+    (fetchurl {
+      name = "0002-remove-executable-bit.patch";
+      url = "https://github.com/NationalSecurityAgency/ghidra/commit/e2a945624b74e5d42dc85e9c1f992315dd154db1.patch";
+      sha256 = "sha256-DUePl8sZOhouivkrOkRzg4pmnglTxE9FVOxozvfhfPc=";
+    })
```

Like you discovered, it seems like `fetchpatch` has [issues](https://github.com/NixOS/nixpkgs/issues/266556), and fetchurl is indeed the way to go. To minimize the chance of unreproducibility however i think we should use the `diff` version instead.

```suggestion
    # we use fetchurl since the fetchpatch normalization strips the whole diff
    # https://github.com/NixOS/nixpkgs/issues/266556
    (fetchurl {
      name = "0002-remove-executable-bit.patch";
      url = "https://github.com/NationalSecurityAgency/ghidra/commit/e2a945624b74e5d42dc85e9c1f992315dd154db1.diff";
      sha256 = "sha256-something";
    })
```